### PR TITLE
migrated the spotify connector to use dscc-gen.

### PR DESCRIPTION
**Pull request by diminishedprime**

### CODE REVIEW

File: spotify/scripts/dscc-scripts.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      throw new Error(`${args.script} is not supported yet.`);
+    }
+  }
+};
+
+(async () => {
+  try {
+    await main();
+  } catch (e) {
+    console.error(`${e}`);
```
Comment (Lucretiel) [inline spotify/scripts/dscc-scripts.js]: This should just be `console.error(e)`

File: spotify/scripts/dscc-scripts.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+const updateDeployment = async (deploymentId) => {
+  await exec(`npx @google/clasp deploy --deploymentId ${deploymentId}`);
+};
+
+const main = async () => {
+  switch (args.script) {
+    case 'try_production': {
+      return await openDeployment(
```
Comment (Lucretiel) [inline spotify/scripts/dscc-scripts.js]: You can replace all of these `return await` with just `return`

File: spotify/scripts/dscc-scripts.js (hunk 1 above)
Comment (Lucretiel) [inline spotify/scripts/dscc-scripts.js]: This should just be `console.error(e)`

File: spotify/scripts/dscc-scripts.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  const green = chalk.rgb(15, 157, 88);
+  const deploymentUrl = `https://datastudio.google.com/datasources/create?connectorId=${deploymentId}`;
+  const formattedUrl = green(
+    terminalLink(`${deploymentName} deployment`, deploymentUrl)
+  );
+  console.log(`Opening: ${formattedUrl}`);
+  return open(deploymentUrl);
+};
+
+const updateDeployment = async (deploymentId) => {
```
Comment (Lucretiel) [inline spotify/scripts/dscc-scripts.js]: Does this function need to be async? Could it just be `(deploymentId) => exec(...)`

File: spotify/scripts/dscc-scripts.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+const open = require('opn');
+const argparse = require('argparse');
+const packageJson = require('../package.json');
+const cp = require('child_process');
+const chalk = require('chalk');
+const terminalLink = require('terminal-link');
+
+const exec = (command, options) => {
+  return new Promise((resolve, reject) => {
+    const child = cp.exec(command, options, (err, stdout, stderr) => {
```
Comment (Lucretiel) [inline spotify/scripts/dscc-scripts.js]: `exec` is a serious code smell, as it's a common source of injection attacks. Even if you think it's okay in this particular case, you should strongly prefer `execFile` unless you have a *really* compelling reason not to.

File: spotify/scripts/dscc-scripts.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+const parser = new argparse.ArgumentParser({
+  version: '1.0',
+  addHelp: true,
+  description: 'Scripts for managing connectors.'
+});
+
+parser.addArgument(['-s', '--script'], {
+  choices: ['try_production', 'update_production', 'try_latest'],
+  dest: 'script',
```
Comment (Lucretiel) [inline spotify/scripts/dscc-scripts.js]: I think `dest` is optional, it picks it out from `--script` if it uses the same behavior as Python argparse

File: spotify/scripts/dscc-scripts.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  help: 'The script to run.',
+  required: true
+});
+
+const args = parser.parseArgs();
```
Comment (Lucretiel) [inline spotify/scripts/dscc-scripts.js]: This should happen in `main`, not globally

File: spotify/scripts/dscc-scripts.js (hunk 4 above)
Comment (Lucretiel) [reply spotify/scripts/dscc-scripts.js]: In particular, this function should have this interface:

    await exec(['npx', '@google/clasp', 'deploy', ...], {options}).then(...)

File: spotify/scripts/dscc-scripts.js (hunk 2 above)
Comment (diminishedprime) [reply spotify/scripts/dscc-scripts.js]: New code is in https://github.com/googledatastudio/dscc-scripts/blob/master/src/connector.ts and removed awaits from the switch block.

File: spotify/scripts/dscc-scripts.js (hunk 1 above)
Comment (diminishedprime) [reply spotify/scripts/dscc-scripts.js]: I'll update that in the template.

File: spotify/scripts/dscc-scripts.js (hunk 4 above)
Comment (diminishedprime) [reply spotify/scripts/dscc-scripts.js]: The new version uses execa, which is using execFile as suggested here.

File: spotify/scripts/dscc-scripts.js (hunk 6 above)
Comment (diminishedprime) [reply spotify/scripts/dscc-scripts.js]: This is fixed in the newer version.

**Comment by diminishedprime** [comment]
Thanks for the review. After this (and other outstanding PRs) gets merged in, I'll switch to using @google/dscc-scripts as a dependency which addresses most of your feedback here.

**Pull request merged**